### PR TITLE
feat: allow setting User-Agent header

### PR DESCRIPTION
**Pull request by jotatoledo**
* Add a method to set the User-Agent header of the
underlying HttpClient instance.

### CODE REVIEW

**Comment by jotatoledo** [comment]
CI error seems to be unrelated

**Comment by mtrdp642** [comment]
CI error is unrelated.

Is there an error or issue you're seeing when the user agent isn't set? As a side note: I would prefer to see the setting of the user agent header value in a constructor overload rather than in it's own method as I can't think of a reason why a user would need to change the user agent value after client initialization. What are your thoughts?

**Comment by jotatoledo** [comment]
Is listead as one requirement in the (fair use policy)[https://pokeapi.co/docs/v2.html] of PokeAPI. I agree with the constructor overload, I went with this approach as it has less impact on the existent API :P

Ill refactor later

**Comment by jotatoledo** [comment]
@mtrdp642 refactored, please review

File: PokeApiNet/Data/PokeApiClient.cs [hunk 1]
```diff
@@ -44,6 +79,12 @@ public void Dispose()
             _client.Dispose();
         }
 
+        private static ProductHeaderValue GetDefaultUserAgent()
+        {
+            var version = typeof(PokeApiClient).Assembly.GetName().Version;
+            return new ProductHeaderValue("PokeApiNet", $"{version.Major}.${version.Minor}");
```
Comment (mtrdp642) [inline PokeApiNet/Data/PokeApiClient.cs]:
```suggestion
            return new ProductHeaderValue("PokeApiNet", $"{version.Major}.{version.Minor}");
```

File: PokeApiNet/Data/PokeApiClient.cs [hunk 2]
```diff
@@ -23,17 +28,48 @@ public class PokeApiClient : IDisposable
         /// Default constructor
         /// </summary>
         public PokeApiClient()
+            : this(DefaultUserAgent)
+        {
+        }
+
+        /// <summary>
+        ///
```
Comment (mtrdp642) [inline PokeApiNet/Data/PokeApiClient.cs]: Can you add a docs blurb for this constructor, please?

**Pull request merged**